### DISCUSSION TITLE
telnet_login param

### DISCUSSION THREAD

**Discussion by yielditercode** [ktbyers/netmiko#3008] [Ideas]
hi, ktbyers:
     (netmiko == 4.1.2)
     1. I try login in device and with default params, but sometime device return data with 15s or longer( device config with 3A and 3A not accept it). I rewirte telnet_login and modity delay_factor = 10 and max_loops = 20 to read_channel, it login successfully。
     Should the future version accept input params? 
     2. When I try to run rollback with device(use send_command_timing),  read_channel_timing's loop_delay is 0.1, it so quickly to read data, sometime channel not return anything
     Should the future version accept dynamic input param ?

     
thanks!

**Comment by ktbyers** [comment]
Yeah, telnet_login is a bit of a mess and likely needs improved / needs rewritten. Setting delay_factor=10 doesn't sound like the solution (however on the face of it).

I don't think input parameters would help as in most cases people do not invoke this i.e. the call is embedded in other places in Netmiko code. It would probably be better if it was wired to the most logical timeout that you could pass in via ConnectHandler arguments.

`send_command_timing()` behavior you should be able to adjust using read_timeout and last_read. I would probably try setting last_read argument to 4 seconds or if that doesn't work then to 8 seconds.
